### ISSUE TITLE
Feature Request: Implement yamllint rules

### ISSUE THREAD

**Issue by roaldnefs** [warpnet/salt-lint#240]
**Is your feature request related to a problem? Please describe.**
Allow running the same rules as [yamllint](https://github.com/adrienverge/yamllint) on SLS files to check for issues such as indentation, etc.

**Describe the solution you'd like**
Add rule to run [yamllint](https://github.com/adrienverge/yamllint) on SLS files after stripping all Jinja expressions, statements and comments.  See the [YamlLintRule](https://github.com/ansible-community/ansible-lint/blob/master/src/ansiblelint/rules/YamllintRule.py) in ansible-lint as an example.

**Describe alternatives you've considered**
Re-implementing all the yamllint rules separately.

**Additional context**
This feature request duplicates #106, #105, #92.

**Comment by colttt** [comment]
any news on that?
maybe you can add this to you salt-lint docker file to install?!

**Comment by jbouter** [comment]
Hi @colttt !

It's not as easy as simply adding yamllint. But for now we don't have a status update sadly.

Feel free to use our container image for salt-lint and add yamllint yourself though. Or implement a new step in your CI where you run yamllint on all sls files. Though be prepared for a whole lot of false positives.

**Comment by colttt** [comment]
> Feel free to use our container image for salt-lint and add yamllint yourself though.
I try that, but I can't install it, permission denied with the following config (as you remember, its my first time ;) )

```
default:
 before_script:
    - apk add yamllint

```
I got:
```
$ apk add yamllint
ERROR: Unable to lock database: Permission denied
ERROR: Failed to open apk database: Permission denied
```

**Comment by jbouter** [comment]
It's a hardened image. You can't just add stuff to the container itself.

For the [Dockerfile](https://github.com/warpnet/salt-lint/blob/main/docker/Dockerfile) and build it yourself.

**Comment by roaldnefs** [comment]
Opened [#297 ](https://github.com/warpnet/salt-lint/discussions/297) to further discuss to possible solutions.

This issues related to #246 and #258.
